cryptography/src/handshake.rs

```diff
@@ -184,21 +185,25 @@ pub struct Context<S, P> {
     ok_timestamps: Range<u64>,
     my_identity: S,
     peer_identity: P,
+    app_namespace: Vec<u8>,
 }
 
 impl<S, P> Context<S, P> {
     /// Creates a new handshake context.
-    pub const fn new(
+    #[allow(clippy::missing_const_for_fn)]
+    pub fn new(
         current_time_ms: u64,
         ok_timestamps: Range<u64>,
         my_identity: S,
         peer_identity: P,
+        app_namespace: Vec<u8>,
     ) -> Self {
         Self {
             current_time: current_time_ms,
             ok_timestamps,
             my_identity,
             peer_identity,
+            app_namespace,
         }
     }
 }
```

Comment:
Great idea!

The change I'd like to see is to:

- take in an `&Transcript` as the first argument,
- store `Transcript` inside of `Context`,
- initialize this by calling `fork` on the first argument, using the file namespace as the string.

Then stream would pass in `&Transcript::new(config.namespace)`